src/coreclr/jit/redundantbranchopts.cpp

```diff
@@ -1588,6 +1595,29 @@ bool Compiler::optJumpThreadCore(JumpThreadInfo& jti)
         }
     }
 
+    // If this is a phi-based threading, and the block we're bypassing has
+    // a memory phi, and the new successors do not, mark the block with BBF_ALTERED_MEMORY_PHI
```

Comment:
I'm guessing `BBF_ALTERED_MEMORY_PHI` was an earlier name of `BBF_NO_CSE_IN`?

Which I suppose ties in to my other question: technically this only needs to kill memory CSEs, is that correct?  Presumably this is just simpler for the interim fix.

Reply:
Ah., thanks for spotting that.

Yes, it could just kill memory CSEs, but it is actually tricky to figure out if a VN is memory dependent. If we could do that then we could instead kill memory CSEs in blocks with "memory havoc" and fix the bug a bit more surgically.